### ISSUE TITLE
Nick completion cycling does not work unless the nick is in the beginning of the line

### ISSUE THREAD

**Issue by lorenzhs** [glowing-bear/glowing-bear#1062]
When completing a nickname, GB suggests the user who spoke most recently first, and then cycles through the remaining users based on when they spoke last. However, this appears to be broken unless the nick to be completed is the first thing in the line. I can't cycle through the possible matches if it's in the middle or at the end of the line. I think this used to work, making this a possible regression that we should fix.

**Comment by AStove** [comment]
I checked to see if I also had this and I did! I then ran it locally to fix it and the bug dissapeared. Then ran it on the original app and didn't have it there anymore either. Any special conditions to this?

**Comment by lorenzhs** [comment]
I haven't really attempted to track this down yet, so no idea, sorry :/

**Comment by AStove** [comment]
Could this be a duplicate of #901 ?

**Comment by lorenzhs** [comment]
Probably not. #901 is that <tab> does nothing except on the first line, this issue is that cycling of completion suggestions only works at the beginning of the line. They're related and could well have similar fixes, but I don't think they're the same issue, as different broken behaviour is exhibited.

**Comment by AStove** [comment]
As of version 2.4 of weechat the default value of weechat.completion.nick_completer is ": " this is what is breaking it.

This works correctly with weechat.completion.nick_completer set to ":" which used to be the default value before 2.4.

**Comment by AStove** [comment]
For future reference, this one can be fixed once Weechat 2.9 is used more using: #1122

**Comment by lorenzhs** [comment]
yes but as always, requiring a round-trip to weechat for something that we could do without one makes high-latency connections even more annoying than they already are
